File: PitBoard(Android)/app/src/main/java/com/pitboard/app/schedule/RaceScheduleSource.kt
package com.pitboard.app.schedule

import com.pitboard.app.data.EventEntity
import com.pitboard.app.data.RaceSeries

/**
 * Contrato que implementan las fuentes de calendario, una por serie (ver RaceSeries) — mismo
 * patrón que StandingsSource para las clasificaciones. El repositorio no sabe ni le importa si
 * detrás hay JSON o HTML, solo pide la lista de sesiones ya normalizadas.
 */
interface RaceScheduleSource {
    val series: RaceSeries

    suspend fun fetch(): List<EventEntity>
}

File: PitBoard(Iphone)/PitBoardKit/Sources/PitBoardKit/Schedule/RaceScheduleSource.swift
import Foundation

/// Contrato que implementan las fuentes de calendario, una por serie — equivalente
/// exacto de `RaceScheduleSource.kt`. Mismo patrón que `StandingsSource` para las
/// clasificaciones: el repositorio no sabe ni le importa si detrás hay JSON o HTML, solo
/// pide la lista de sesiones ya normalizadas.
public protocol RaceScheduleSource: Sendable {
    var series: RaceSeries { get }

    func fetch() async throws -> [EventDraft]
}
